src/lib.rs

```diff
@@ -73,4 +86,38 @@ where
         self.state = s >> 2;
         Ok(s.into())
     }
+
+
+    #[cfg(feature = "table-decoder")]
+    /// Call `update` to evaluate the next state of the encoder, propagates errors from `InputPin` read
+    pub fn update(&mut self) -> Result<Direction, Either<A::Error, B::Error>> {
+        // Implemented after https://www.best-microcontroller-projects.com/rotary-encoder.html
+        self.prev_next <<= 2;
+        if self.pin_a.is_high().map_err(Either::Left)? {
+            self.prev_next |= 0x02;
```

Comment:
I find the binary a lot easier to read and figure out what's going on, can we write these as `0b0010` and `0b0001`?